### PR TITLE
feat: add an option to make the path `shortened`

### PR DESCRIPTION
**Pull request by B4mbus**
Added an option to make the path shortened.
image

### CODE REVIEW

**Comment by ghillb** [comment]
Hey, this looks cool, and I'm inclined to merge it with some minor changes. Are you OK with implementing some suggestions, I make, and we merge it then?

**Comment by B4mbus** [comment]
Ofc! Glad you like it.

File: lua/cybu/init.lua [hunk 1]
```diff
@@ -46,6 +92,12 @@ cybu.get_bufs = function()
       name = string.gsub(name, cwd_path, "")
     elseif c.opts.style.path == v.style_path.tail then
       name = vim.fn.fnamemodify(name, ":t")
+    elseif c.opts.style.path == v.style_path.shortened then
+      local preffered_separator = get_preffered_path_separator()
+      local alternate_separator = get_alternate_separator(preffered_separator)
+      local normalized_name = vim.fn.fnamemodify(name, ':p'):gsub(alternate_separator, preffered_separator)
```
Comment (ghillb) [inline lua/cybu/init.lua:98]: Could you explain why this has to be done this way (alternate_separator / preffered_separator)?

File: lua/cybu/init.lua [hunk 2]
```diff
@@ -6,6 +6,52 @@ local infobar = require("cybu.infobar")
 local cybu, _state = {}, {}
 local has_plenary, strings = pcall(require, "plenary.strings")
 
+local function get_preffered_path_separator()
```
Comment (ghillb) [inline lua/cybu/init.lua:9]: `preffered` -> `preferred`.

Comment (ghillb) [inline lua/cybu/init.lua:9]: I think `utils.lua` is a better place for all the new functions.

**Comment by ghillb** [comment]
Great! I would like to synchronize the looks with the `shorten` option of [telescope](https://github.com/nvim-telescope/telescope.nvim/blob/c92f86386f8446e4deaa79941baabaf825683be9/lua/telescope/config.lua#L302).
Here are pics for comparison:
telescope
cybu

Notable differences, telescope (upper image) only uses the relative paths, when `shorten` is set, adds the first char after an `.` for a dot directory and per default the `/` separator is displayed on Linux.

To keep the combination absolute + shorten, one could implement the shorten functionality under a new option key, e.g., `style = { shorten_path = true/false }` or  `style = { path_abbreviation = 'shorten'/'default' }`, to make it easier to implement another abbreviation variant (like telescope's `'smart'`) in the future.

I think the last one is a bug, `vim.fn.has('win32')` returns "0" on Linux, which is truthy in Lua.

**Review by ghillb — CHANGES_REQUESTED** [review]

File: lua/cybu/init.lua [hunk 3]
```diff
@@ -6,6 +6,52 @@ local infobar = require("cybu.infobar")
...
+  if vim.fn.has('win32') then
+    return '\\'
+  else
+    return '/'
+  end
+end
+
+local function get_alternate_separator(sep)
```
Comment (ghillb) [inline lua/cybu/init.lua:17]: Maybe the plenary path lib is more robust for this kind of stuff, but that's just a consideration:

https://github.com/nvim-lua/plenary.nvim/blob/master/lua/plenary/path.lua

https://github.com/nvim-telescope/telescope.nvim/blob/6bddc38c25af7b50f99cb0c035248d7272971810/lua/telescope/utils.lua#L19

https://github.com/nvim-telescope/telescope.nvim/blob/6bddc38c25af7b50f99cb0c035248d7272971810/lua/telescope/utils.lua#L254

**Comment by B4mbus** [comment]
It'd be lovely if you could try that PR locally because I'm not in a position right now to try it on linux, but I believe it should work no problem.

**Comment by ghillb** [comment]
Yes, I will do it soon.

Regarding 1: It is probably a minor misunderstanding, what I meant, was to keep the current `style = {path = "relative",            -- absolute, relative, tail}` option and to introduce a new option (e.g.: `style = { path_abbreviation = 'shorten'/'default' }`) for the new functionality.
This yields the possibility to shorten relative AND absolute paths, and chance to introduce new abbreviation options easier in the future without breaking changes and could be probably done in relatively few lines of code.

**Comment by B4mbus** [comment]
Oh, that actually makes sense. How about instead of `path_abbrievation = 'shorten'/'default'` - `path_abbrievation =   'shorten'/'none'` or even `shorten_path = true/false`?

**Comment by ghillb** [comment]
`path_abbrevation =   'shorten'/'none'` sounds good to me (in case there will be another abbreviation scheme in the future, this will prevent breaking changes)

**Comment by ghillb** [comment]
> It'd be lovely if you could try that PR locally because I'm not in a position right now to try it on linux, but I believe it should work no problem.

Yes, it works fine now on Linux, and the improvements you made look good too.

**Comment by B4mbus** [comment]
It should be good now.

File: lua/cybu/config.lua [hunk 4]
```diff
@@ -13,6 +13,7 @@ local default_config = {
   },
   style = {
     path = "relative",
+    path_abbrievation = "none",
```
Comment (ghillb) [inline lua/cybu/config.lua:16]:
```suggestion
    path_abbreviation = "none",
```

File: lua/cybu/init.lua [hunk 5]
```diff
@@ -41,12 +41,20 @@ cybu.get_bufs = function()
...
+    -- adjust buf names
+    if c.opts.style.path == v.style_path.absolute then
+      name = vim.fn.fnamemodify(name, ':p')
+    elseif c.opts.style.path == v.style_path.relative then
       name = string.gsub(name, cwd_path, "")
     elseif c.opts.style.path == v.style_path.tail then
       name = vim.fn.fnamemodify(name, ":t")
     end
+
+    if c.opts.style.path_abbrievation == v.style_path_abbrievation.shortened then
```
Comment (ghillb) [inline lua/cybu/init.lua:54]:
```suggestion
    if c.opts.style.path_abbreviation == v.style_path_abbreviation.shortened then
```

File: lua/cybu/vars.lua [hunk 6]
```diff
@@ -13,6 +13,7 @@ vars.behavior = {
   },
 }
 vars.style_path = { absolute = "absolute", relative = "relative", tail = "tail" }
+vars.style_path_abbrievation = { shortened = "shortened", none = "none" }
```
Comment (ghillb) [inline lua/cybu/vars.lua:16]:
```suggestion
vars.style_path_abbreviation = { shortened = "shortened", none = "none" }
```

**Review by ghillb — CHANGES_REQUESTED** [review]
I've made only some minor suggestions for docs and typos. Otherwise, LGTM, and I will merge soon :)

**Comment by B4mbus** [comment]
Fixed it.

But oof, don't mind the commit name, I changed them to `abbreviation`, not `abbrevation`, just a typo in the commit :(.

File: lua/cybu/init.lua [hunk 7]
```diff
@@ -41,12 +41,20 @@ cybu.get_bufs = function()
...
+    -- adjust buf names
+    if c.opts.style.path == v.style_path.absolute then
+      name = vim.fn.fnamemodify(name, ':p')
+    elseif c.opts.style.path == v.style_path.relative then
       name = string.gsub(name, cwd_path, "")
     elseif c.opts.style.path == v.style_path.tail then
       name = vim.fn.fnamemodify(name, ":t")
     end
+
+    if c.opts.style.path_abbreviation == v.style_path_abbrevation.shortened then
```
Comment (ghillb) [inline lua/cybu/init.lua:54]:
```suggestion
    if c.opts.style.path_abbreviation == v.style_path_abbreviation.shortened then
```

**Review by ghillb — COMMENTED** [review]
All good, just missed one (see suggestion)

**Comment by B4mbus** [comment]
Oh man! Right. Really sorry, I'm a bit dizzy. Hope it's all good now, thanks for your suggestions.

**Pull request merged**